feat(BlogHighlightSection, EbookHero, Resources): make components dynamic and fetch data from API

mahdisolgi1 opened the pull request:
### BlogHighlightSection:

- Made the BlogHighlightSection component dynamic to render real data.
- Updated the hooks to fetch complete data for displaying all article details.


### EbookHero:

- Updated the EbookHero component to fetch data from the API.
- Dynamically rendered the fetched data in the component.

### Resources:


- Updated the Resources component to fetch data from the API.
- Dynamically rendered the fetched data in the component.



This PR enhances the flexibility of the components by replacing static content with dynamic data fetched from the API. It improves the overall structure and ensures that the components reflect real-time data when rendered.

mahdisolgi1 committed:
feat: add design system and create static mobile layout

- Implemented design system with typography, colors, and spacing
- Created static version of the mobile layout following the design specs

mahdisolgi1 committed:
feat(Resources): fetch and display dynamic data in Resources component
- Updated Resources component to fetch data from API
- Rendered fetched data dynamically in the component

mahdisolgi1 committed:
feat(EbookHero): fetch and display dynamic data in EbookHero component
- Updated EbookHero component to fetch data from API
- Rendered fetched data dynamically in the component

mahdisolgi1 committed:
feat(BlogHighlightSection): make component dynamic and enhance hooks for article

- Made BlogHighlightSection dynamic to render real-data
- Updated hooks to fetch complete data for displaying all articles' details

mahdisolgi1 committed:
fix(EbookHero, Resources): add missing alt props to images

- Ensured all images have descriptive alt text in EbookHero and Resources components for improved accessibility

mahdisolgi1 committed:
chore: convert all CSS files to SCSS

- Changed all `.css` files to `.scss` to enable SCSS features
- Updated file imports to work properly

mahdisolgi1 committed:
feat: add mixin.scss and update vite.config.ts for global imports

- Created a global mixin.scss file containing reusable mixins for breakpoints.
- Updated vite.config.ts to include mixin.scss globally, eliminating the need to import it in each file manually.

mahdisolgi1 committed:
feat(TopBanner): make component responsive for laptop screens

- Updated TopBanner component to be responsive on laptop devices.
- Applied layout adjustments and styling changes for optimal display on laptop screens.

amirrezaDev1378 commented on backend/.env.example:
restore this file

amirrezaDev1378 commented on bash.exe.stackdump:
add all executables and stackdump files to gitignore

amirrezaDev1378 commented on frontend/package.json:
use a stable version of eslint-plugin-react-hooks

amirrezaDev1378 commented on frontend/public/Ali-Smith.png:
Where is this for? shouldn't this be dynamically fetched from backend?

amirrezaDev1378 commented on frontend/public/Star.svg:
use react icons for icons

amirrezaDev1378 commented on frontend/public/arrow-to-up-right-black.svg:
use react icons for icons

amirrezaDev1378 commented on frontend/public/arrow-to-up-right.svg:
use react icons for icons

amirrezaDev1378 commented on frontend/public/heart-icon.svg:
use react icons for icons

amirrezaDev1378 commented on frontend/public/linkdin-icon.svg:
use react icons for icons

amirrezaDev1378 commented on frontend/public/share-icon.svg:
use react icons for icons

amirrezaDev1378 commented on frontend/src/components/BlogHighlightSection.module.css:
no need for fallback values if you have all ready set the color variable

amirrezaDev1378 commented on frontend/src/components/BlogHighlightSection.tsx:
```tsx
            {" "}
                              {article?.shares > 1000
                                 ? `${
                                      article.shares % 1000 === 0
                                         ? article.shares / 1000
                                         : (article.shares / 1000).toFixed(1)
                                   }k`
                                 : article?.shares}
```

this can be simplified

amirrezaDev1378 commented on frontend/src/components/Button.module.css:
add transition for any changes that affect the UI like in this case the background color

amirrezaDev1378 commented on frontend/src/components/EbookHero.module.css:
```diff
@@ -0,0 +1,326 @@
+   display: flex;
+   flex-direction: column;
+   justify-content: center;
+   align-items: flex-start;
```
duplicate styles

amirrezaDev1378 commented on frontend/src/components/EbookHero.module.css:
```diff
@@ -0,0 +1,326 @@
+}
+.ebookHeroImg {
+   border-radius: 1rem;
+   /* height: 16.9rem; */
```
remove comment if there is no need for it

amirrezaDev1378 commented on frontend/src/components/EbookHero.tsx:
```diff
@@ -0,0 +1,188 @@
+   const res = data?.data?.data;
+
+   const fourRandomAuthors = getRandomObjects(res, 4);
+   console.log(fourRandomAuthors);
```
remove log

amirrezaDev1378 commented on frontend/src/components/TopBanner.tsx:
```diff
@@ -0,0 +1,14 @@
+   return (
+      <section className={styles.topBannerSection}>
+         <p className={styles.topBannerP}>Subscribe to our Newsletter For Blogs and Resources</p>
+         {/* <ArrowToUpRight width={24} height={24} fill="currentColor" /> */}
```
remove comment

amirrezaDev1378 commented on frontend/src/helpers/getRandomObjects.ts:
```diff
@@ -0,0 +1,10 @@
+export default function getRandomObjects<T>(array: T[] | undefined, count: number): T[] {
+   if (!array || array.length === 0) {
```
use this instead

!(array?.length > 0)

amirrezaDev1378 commented on frontend/src/helpers/getRandomObjects.ts:
this code can be optimized but it's not that important

amirrezaDev1378 commented on frontend/src/hooks/useDarkMode.tsx:
```diff
@@ -0,0 +1,24 @@
+const useDarkMode = (): [boolean, () => void] => {
+   const [darkMode, setDarkMode] = useState<boolean>(() => {
+      const savedMode = localStorage.getItem("darkMode");
+      return savedMode ? JSON.parse(savedMode) : false;
```
put this out of use-state on its own variable, and also either don't use JSON or if you do handle any errors by falling back to the default theme

mahdisolgi1 committed:
feat(Nav): improve responsiveness on laptop screens

- Updated Nav component layout for better responsiveness on laptop screens
- Adjusted styles to optimize navigation bar appearance on larger screens

mahdisolgi1 committed:
chore: restore backend/.env.example file

mahdisolgi1 committed:
chore: add stack dump and executable files to .gitignore

mahdisolgi1 committed:
chore: use stable version of eslint-plugin-react-hooks (v5.0.0)

mahdisolgi1 committed:
chore: delete unnecessary image files in public folder

mahdisolgi1 committed:
feat: Replace all icons with React Icons library

-Replaced existing icons throughout the project with new icons from the React Icons library for improved performance and consistency.

mahdisolgi1 committed:
chore: remove unused icons from public directory

mahdisolgi1 committed:
Remove fallback values for variables across the project

- Deleted all fallback values for variables to streamline code and improve clarity.
- Updated relevant documentation where necessary.

mahdisolgi1 committed:
refactor(BlogHighlightSection): simplify code for count formatting

-Optimized the code for displaying share count in the BlogHighlightSection component by consolidating logic for formatting numbers over 1000, reducing redundancy.

mahdisolgi1 committed:
delete the duplicate style in EbookHero styles

mahdisolgi1 committed:
Refactor dark mode theme handling:

- Moved theme retrieval from localStorage outside of useState to a separate function (`getInitialTheme`).
- Added error handling for localStorage access and JSON parsing. In case of an error, the theme falls back to the default light mode.

mahdisolgi1 committed:
Fix getRandomObjects function

mahdisolgi1 committed:
Clean up codebase: Removed unnecessary comments throughout the project

mahdisolgi1 committed:
Remove all console.log statements from the project

mahdisolgi1 committed:
feat(Button): Add smooth transitions and hover effects

- Added CSS transitions for background color and transform effects.
- Implemented hover state to change background color and apply a slight scale transformation.
-added some classname to Button component
.

mahdisolgi1 committed:
fix a small bug in the button component

mahdisolgi1 committed:
use the Button componnent inside the navbar

mahdisolgi1 committed:
fix a small bad styling in the Button styles

mahdisolgi1 commented:
Fixed errors mentioned in the PR comment.

amirrezaDev1378 reviewed (approved):
LGTM!
